### PR TITLE
bitcoin 27.0

### PR DESCRIPTION
**Pull request by dnhn**
Created with `brew bump-formula-pr`.

### CODE REVIEW

File: Formula/b/bitcoin.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Bitcoin < Formula
   desc "Decentralized, peer to peer payment network"
   homepage "https://bitcoincore.org/"
-  url "https://bitcoincore.org/bin/bitcoin-core-26.1/bitcoin-26.1.tar.gz"
-  sha256 "9164ee5d717b4a20cb09f0496544d9d32f365734814fe399f5cdb4552a9b35ee"
+  url "https://bitcoincore.org/bin/bitcoin-core-27.0/bitcoin-27.0.tar.gz"
+  sha256 "9c1ee651d3b157baccc3388be28b8cf3bfcefcd2493b943725ad6040ca6b146b"
   license "MIT"
   head "https://github.com/bitcoin/bitcoin.git", branch: "master"
```
Comment (fanquake) [inline Formula/b/bitcoin.rb:8]: The two patches being applied should be removed now.

Comment (chenrui333) [reply Formula/b/bitcoin.rb:8]: 👍

**Comment by chenrui333** [comment]
```
==> /home/linuxbrew/.linuxbrew/Cellar/bitcoin/27.0/bin/test_bitcoin
  Running 601 test cases...
  test/serfloat_tests.cpp(95): error: in "serfloat_tests/double_serfloat_tests": check TestDouble(std::numeric_limits<double>::denorm_min()) == 0x1 has failed [0 != 1]
  test/serfloat_tests.cpp(96): error: in "serfloat_tests/double_serfloat_tests": check TestDouble(-std::numeric_limits<double>::denorm_min()) == 0x8000000000000001 has failed [9223372036854775808 != 92233720368[54](https://github.com/Homebrew/homebrew-core/actions/runs/8726157834/job/23940840932#step:4:55)775809]
  
  *** 2 failures are detected in the test module "Bitcoin Core Test Suite"
  
  Error: bitcoin: failed
```

- previous linux test report, https://github.com/bitcoin/bitcoin/issues/28941

**Comment by fanquake** [comment]
@chenrui333 is there something special/different about the brew Linux build environments? We are not seeing failures like this in any other Linux environment or any of our own CI.

The patches dropped here were relaxations to the tests, shipped with 27.0, however it looks like that still wasn't quite enough.

**Comment by chenrui333** [comment]
@fanquake let's see how it looks like with a rerun.

**Comment by fanquake** [comment]
The macOS failure looks like it could be a brew issue. Maybe needs a rebase? The Linux failure is:
```bash
  Running 601 test cases...
  test/serfloat_tests.cpp(95): error: in "serfloat_tests/double_serfloat_tests": check TestDouble(std::numeric_limits<double>::denorm_min()) == 0x1 has failed [0 != 1]
  test/serfloat_tests.cpp(96): error: in "serfloat_tests/double_serfloat_tests": check TestDouble(-std::numeric_limits<double>::denorm_min()) == 0x8000000000000001 has failed [9223372036854775808 != 9223372036854775809]
  
  *** 2 failures are detected in the test module "Bitcoin Core Test Suite"
```

I'll reopen https://github.com/bitcoin/bitcoin/issues/28941 and we can followup.

File: Formula/b/bitcoin.rb [hunk 2]
```diff
@@ -46,16 +46,6 @@ class Bitcoin < Formula
     cause "Requires std::filesystem support"
```
Comment (fanquake) [inline Formula/b/bitcoin.rb:46]: Note that we now also require GCC 10+, due to C++20 usage, so this should also be updated.

**Comment by fanquake** [comment]
@chenrui333 I have a commit here, https://github.com/fanquake/homebrew-core/commit/dccc9aaf0a0b006a80d221b9de7d18bbb4c1bc0f, that can be used, to unblock this. Happy for someone to pull that in / open a PR.

**Comment by fanquake** [comment]
Note that 27.1 is now available, so I've opened #174760.

**Comment by chenrui333** [comment]
27.1 seems working for us. 💯

**Comment by fanquake** [comment]
Should be closable, now that #174775 is merged.

**Pull request closed without merging**